tests/v3/test_api.py

```diff
@@ -117,6 +117,10 @@ def test_open_with_mode_r_plus(tmp_path: pathlib.Path) -> None:
 
 
 def test_open_with_mode_a(tmp_path: pathlib.Path) -> None:
+    # Open without shape argument should default to group
+    g = zarr.open(store=tmp_path, mode="a")
+    assert isinstance(g, Group)
+
     # 'a' means read/write (create if doesn't exist)
     arr = zarr.open(store=tmp_path, mode="a", shape=(3, 3))
     assert isinstance(arr, Array)
```

Comment:
I'm surprised this does not fail after the previous statement. Shouldn't we be raising an error if a Group already exists at this path?

Reply:
You are right, I had some issues with my local testing config and missed this, but have fixed it and now can see the failures.